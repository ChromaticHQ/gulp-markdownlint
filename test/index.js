const gulp = require('gulp')
const path = require('path')
const gulpMarkdownlint = require('../src')

const resolve = glob => path.join(__dirname, 'files', glob)

describe('test index', () => {
  it('check error message', () => {
    expect.assertions(1)
    // eslint-disable-next-line promise/param-names
    return new Promise((resolved) => {
      gulp.src(resolve('base.md'))
        .pipe(gulpMarkdownlint({
          configFile: resolve('markdownlint.json')
        }))
        .on('error', err => {
          expect(err.message.replace(/.*base.md/g, '')).toMatchSnapshot()
          resolved()
        })
    })
  })

  // it('check console message with reporters', () => {
  //   expect.assertions(1)
  //   jest.spyOn(console, 'info')
  //     .mockImplementation(message => {
  //       expect(message.replace(/^.*\n/, '')).toMatchSnapshot()
  //     })

  //   // eslint-disable-next-line promise/param-names
  //   return new Promise((resolved) => {
  //     gulp.src(resolve('base.md'))
  //       .pipe(gulpMarkdownlint({
  //         configFile: resolve('markdownlint.json'),
  //         reporters: [{
  //           formatter: 'verbose',
  //           console: true,
  //           save: 'test/files/log.txt'
  //         }]
  //       }))
  //       .on('error', () => resolved())
  //   })
  // })

  it('check error with failAfterError: false', () => {
    expect.assertions(0)
    // eslint-disable-next-line promise/param-names
    return new Promise((resolved) => {
      gulp.src(resolve('base.md'))
        .pipe(gulpMarkdownlint({
          configFile: resolve('markdownlint.json'),
          failAfterError: false
        }))
        .on('error', err => {
          console.log(err)
          expect(err.message.replace(/.*base.md/g, '')).toMatchSnapshot()
        })
        .on('finish', resolved)
    })
  })

  it('provide config object instead of config file', () => {
    expect.assertions(1)
    // eslint-disable-next-line promise/param-names
    return new Promise((resolved) => {
      gulp.src(resolve('base.md'))
        .pipe(gulpMarkdownlint({
          configFile: false,
          config: {
            MD022: { lines_above: 0, lines_below: 0 }
          }
        }))
        .on('error', err => {
          console.log(err)
          expect(err.message.replace(/.*base.md/g, '')).toMatchSnapshot()
          resolved()
        })
    })
  })
})
