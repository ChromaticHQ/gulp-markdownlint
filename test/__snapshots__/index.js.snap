// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test index check error message 1`] = `
": 1: MD022/blanks-around-headings/blanks-around-headers Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Below] [Context: \\"# 123\\"]
: 2: MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: \\"* 123 *\\"]"
`;

exports[`test index provide config object instead of config file 1`] = `": 2: MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: \\"* 123 *\\"]"`;
